src/arrayMethodJoin.js

```diff
@@ -6,6 +6,35 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let result = '';
+    let spacer;
+
+    switch (separator) {
+      case undefined:
+        spacer = ',';
+        break;
+
+      case null:
+        spacer = 'null';
+        break;
+
+      default:
+        spacer = separator;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
```

Comment:
Ты меняешь переданный массив, чего оригинальный метод join не делает. 
1. [CODE STYLE] - don't mutate original array elements - it will cause unexpected results later on.